src/mjolnir/util.cc

```diff
@@ -277,17 +279,31 @@ bool build_tile_set(const boost::property_tree::ptree& config,
       osm_data.write_to_temp_files(tile_dir);
     }
   }
+  // Construct edges
+  std::map<baldr::GraphId, size_t> tiles;
+  if (start_stage <= BuildStage::kConstructEdges && BuildStage::kConstructEdges <= end_stage) {
+    tiles = GraphBuilder::BuildEdges(config, osm_data, ways_bin, way_nodes_bin, nodes_bin, edges_bin);
+    // Output manifest
+    TileManifest manifest{tiles};
+    manifest.LogToFile(tile_manifest);
+  }
 
   // Build Valhalla routing tiles
   if (start_stage <= BuildStage::kBuild && BuildStage::kBuild <= end_stage) {
     // Read OSMData from files if building tiles is the first stage
     if (start_stage == BuildStage::kBuild) {
+      if (filesystem::exists(tile_manifest)) {
+        tiles = TileManifest::ReadFromFile(tile_manifest).tileset;
+      } else {
+        tiles =
+            GraphBuilder::BuildEdges(config, osm_data, ways_bin, way_nodes_bin, nodes_bin, edges_bin);
```

Comment:
shouldn't this be a `LOG_ERROR("No tile manifest was found")` or something? we dont support optional stages in tile building, every stage is a required link in the chain. since the construct edges stage always creates the manifest we should fail in the build stage if we dont  have it

Reply:
The thought here was that this would make it easier to migrate the downstream pipeline and unblock the merge of this PR (which effectively changes an API). I've added a log line and TODO here so that we follow up and make this required later.